packages/vscode/syntaxes/astro.tmLanguage.json

```diff
@@ -68,25 +68,64 @@
       ]
     },
     "astro:component": {
-      "name": "meta.tag.component.astro",
-      "begin": "(</?)([$A-Z_][^/?!\\s<>]*|[^/?!\\s<>.]+\\.[^/?!\\s<>]+)\\b",
-      "beginCaptures": {
-        "1": {
-          "name": "punctuation.definition.tag.begin.astro"
-        },
-        "2": {
-          "name": "entity.name.tag.astro support.class.component.astro"
-        }
-      },
-      "end": "(/?>)",
-      "endCaptures": {
-        "1": {
-          "name": "punctuation.definition.tag.end.astro"
-        }
-      },
       "patterns": [
         {
-          "include": "#astro:attribute"
+          "name": "meta.tag.component.astro astro.component.raw",
+          "begin": "(<)([$A-Z_][^/?!\\s<>]*|[^/?!\\s<>.]+\\.[^/?!\\s<>]+)(.+is:raw.*?)(>)",
+          "end": "(</)([$A-Z_][^/?!\\s<>]*|[^/?!\\s<>.]+\\.[^/?!\\s<>]+)(?=\\s|/?>)(>)",
+          "beginCaptures": {
+            "1": {
+              "name": "punctuation.definition.tag.begin.astro"
+            },
+            "2": {
+              "name": "entity.name.tag.astro support.class.component.astro"
+            },
+            "3": {
+              "patterns": [
+                {
+                  "include": "#astro:attribute"
+                }
+              ]
+            },
+            "4": {
+              "name": "punctuation.definition.tag.end.astro"
+            }
+          },
+          "endCaptures": {
+            "1": {
+              "name": "punctuation.definition.tag.begin.astro"
+            },
+            "2": {
+              "name": "entity.name.tag.astro support.class.component.astro"
+            },
+            "3": {
+              "name": "punctuation.definition.tag.end.astro"
+            }
+          },
+          "contentName": "source.unknown"
+        },
+        {
+          "name": "meta.tag.component.astro",
+          "begin": "(</?)([$A-Z_][^/?!\\s<>]*|[^/?!\\s<>.]+\\.[^/?!\\s<>]+)\\b",
```

Comment:
This looks unchanged, but just want to confirm that `namespace.component` is properly highlighted as a component?

Reply:
Yep, what gives it the special highlighting is actually the `support.class.component.astro` scope. Since it makes it highlighted like a class (in TSX files, it's `support.class.component.tsx`)